src/parser/SparqlParser.cpp

```diff
@@ -328,22 +245,11 @@ SparqlFilter SparqlParser::parseRegexFilter(bool expectKeyword) {
 template <typename ContextType>
 auto SparqlParser::parseWithAntlr(
     ContextType* (SparqlAutomaticParser::*F)(void),
-    const ParsedQuery& parsedQuery)
-    -> decltype((std::declval<sparqlParserHelpers::ParserAndVisitor>())
-                    .parseTypesafe(F)
-                    .resultOfParse_) {
-  return parseWithAntlr(F, getPrefixMap(parsedQuery));
-}
-
-// ________________________________________________________________________
-template <typename ContextType>
-auto SparqlParser::parseWithAntlr(
-    ContextType* (SparqlAutomaticParser::*F)(void),
-    SparqlQleverVisitor::PrefixMap prefixMap)
+    SparqlQleverVisitor::PrefixMap prefixMap, std::string query)
     -> decltype((std::declval<sparqlParserHelpers::ParserAndVisitor>())
                     .parseTypesafe(F)
                     .resultOfParse_) {
-  sparqlParserHelpers::ParserAndVisitor p{lexer_.getUnconsumedInput(),
+  sparqlParserHelpers::ParserAndVisitor p{std::move(query),
```

Comment:
I think in the meantime we only need "parseCompletely with antlr" or, if this is the only occurence left,remove this function and move the code directly to the `::parse` function.
Especially there's no need for the lexer anymore (except for the FILTERs, but they don't use ANTLR).